stubs/jmespath/@tests/stubtest_allowlist.txt

```diff
@@ -0,0 +1,3 @@
+jmespath.lexer.LexerTokenizeResult
+jmespath.lexer.LexerTokenizeResults
+jmespath.visitor.TreeNode
```

Comment:
We shouldn't add allowlist entries without adding a comment saying why we're ignoring the test failures. Why was stubtest failing for these three objects?

Reply:
Thanks for the review. I've added these objects here because them are useful when writing code that uses the library, so I want to be able to import them with `from jmespath.visitor import TreeNode`. What would be the recommended way to deal with them, must I keep them private?

Reply:
Hmm, yes, I'd rather we gave them names with leading underscores in that case. Stubtest is pointing to a real issue here: since these names don't exist at runtime, attempting to import them at runtime will fail unless you're in an `if TYPE_CHECKING:` block. Using names with leading underscores will help warn end users of the danger.

Reply:
Makes sense, thanks for the review!